managedtenants/bundles/docker_api.py

```diff
@@ -143,8 +145,48 @@ def _image_exists(self, image):
             # https://github.com/app-sre/sretoolbox/blob/master/sretoolbox/container/image.py#L119
             return False
 
-    def validate_image(self, tag):
+    def check_image_size_non_zero(self, tag):
+        """
+        Bundle containers only contain data, so detecting a 0 size is enough
+        to validate we successfully inserted data in the bundle.
+        """
         image = self.client.images.get(tag)
         self.log.debug(image.attrs)
         if image.attrs.get("Size", -1) == 0:
             raise DockerError(f"Built an empty image: {tag}.")
+
+    def extract_file_from_container(self, tag, path):
+        """
+        Creates a temporary container and returns the given file as an
+        io.BytesIO object.
+
+        :tag str: image from which to create the container
+        :path str: path of the file to be extracted from container
+        :returns: in-memory fileobj of the extracted file
+        """
+        container = None
+        try:
+            in_memory_tar = io.BytesIO()
+            container = self.client.containers.create(tag)
+            tar_bytes, _ = container.get_archive(path)
+
+            for tb in tar_bytes:
+                in_memory_tar.write(tb)
+
+            in_memory_tar.seek(0)
+            with tarfile.open(fileobj=in_memory_tar) as tf:
```

Comment:
```suggestion
            container = self.client.containers.create(tag)
            tar_bytes, _ = container.get_archive(path)

            with tarfile.open(fileobj=io.BytesIO(tar_bytes)) as tf:
```

I don't think you need the intermediary object. I would try initializing `BytesIO` with the constructor.

Reply:
it did not like that

```py
Traceback (most recent call last):
  File "/home/sugar/opt/_mtrepos/managed-tenants-cli/.venv/bin/inspect_index_image.py", line 23, in <module>
    iibuilder._validate_sql_catalog(index_image)
  File "/home/sugar/opt/_mtrepos/managed-tenants-cli/managedtenants/bundles/index_builder.py", line 104, in _validate_sql_catalog
    in_memory_db = self.docker_api.extract_file_from_container(
  File "/home/sugar/opt/_mtrepos/managed-tenants-cli/managedtenants/bundles/docker_api.py", line 171, in extract_file_from_container
    with tarfile.open(fileobj=io.BytesIO(tar_bytes)) as tf:
TypeError: a bytes-like object is required, not 'generator'
```